### ISSUE TITLE
wasm-bindgen version issue

### ISSUE THREAD

**Issue by dimitrilw** [fermyon/leptos-spin#47]
I apologize for being a bit of a surface-level Rust user. I am unsure what tweaks to make to get this building so I can start playing with Leptos SSR on Fermyon Spin. Please advise.

## Platform

- MacOS 15.1
- Spin 2.7.0
- rustc 1.82.0
- cargo 1.82.0

## Steps taken

I simply followed the readme.

- one-time prep:
  - `spin templates install --git https://github.com/fermyon/leptos-spin`
  - `cargo install cargo-leptos`
- template setup:
  - `spin new -t leptos-ssr my-leptos-app -a`
  - `cd my-leptos-app`
  - `spin build --up`

## Result

Build command failed due to version-mismatch on wasm-bindgen. Full details in pic.

<img width="1057" alt="image" src="https://github.com/user-attachments/assets/7fffc846-8af7-40a5-a197-5b0b7018897f">

## Fix Option-1 Fails

When trying to update the wasm-bindgen dependency with `cargo update -p wasm-bindgen --precise 0.2.93`, it resulted in:

<img width="1176" alt="image" src="https://github.com/user-attachments/assets/2247af77-36b5-431e-b6a8-0167ad1e152f">

## Fix Option-2 Fails

First, I did a clean walk-through of the "steps taken" section to ensure it was a clean attempt at the second option. When attempting to resolve via updating the binary with `cargo install -f wasm-bindgen-cli --version 0.2.95`, it appeared to update some stuff (pic).

<img width="670" alt="image" src="https://github.com/user-attachments/assets/7ba1bf38-09ff-4789-b586-99e700c6d28e">

However, `spin up --build` led to the same version error as before, shown in the "result" section's image.

**Comment by ogghead** [comment]
Thanks for mentioning this! I ran into this recently, and was able to fix it by creating a branch of cargo-leptos that uses 0.2.95 bindgen -- https://github.com/ogghead/cargo-leptos has the branch.

I will look into upstreaming this today

**Comment by ogghead** [comment]
Actually, it looks like someone else beat me to it 😄 so the version on their GitHub should work for you (I suspect). Can you try installing it with `cargo install --git https://github.com/leptos-rs/cargo-leptos --locked cargo-leptos` and see if that gets your build working?

**Comment by dimitrilw** [comment]
## TLDR: Worked. Thank you!

(but not closing issue; see bottom)

## Long version:

It was a bit of a wild ride and I admit deep unfamiliarity with Rust and a bit of uncertainty why it appeared to run full recompilation multiple times.

I started with a clean template:

<img width="673" alt="image" src="https://github.com/user-attachments/assets/3878663f-ffff-474d-af3c-df32aa998271">

Then I issued the suggested command (`cargo install --git https://github.com/leptos-rs/cargo-leptos --locked cargo-leptos `), which appeared to compile all of the template's dependencies:

<img width="1244" alt="image" src="https://github.com/user-attachments/assets/1ad80075-400e-4373-8acd-c0d74a713f49">

After that finished, I ran `spin up --build`, which appeared to compile things three times.

<img width="738" alt="image" src="https://github.com/user-attachments/assets/d8374062-0ebe-4d3f-b3e5-c0a708a29f90">

<img width="1123" alt="image" src="https://github.com/user-attachments/assets/e8d93307-f5a5-4d0f-989f-8290f5a5e8b8">

On the third pass, there were a ton of duplicate errors and the "look out, wasi going away" warning:

<img width="627" alt="image" src="https://github.com/user-attachments/assets/32bcfea8-4f7f-45d9-80cb-0243c86d7fe5">

In the end, it finished, started serving, and was saving button clicks to the SQLite DB.

<img width="569" alt="image" src="https://github.com/user-attachments/assets/8167e32e-56b2-44c6-9094-2bcbe7909243">

<img width="939" alt="image" src="https://github.com/user-attachments/assets/ca2f6ba6-c71b-4625-a304-917c64eace7d">

I do not understand why the issuance of `spin up --build` appeared to have three cycles of compilation, the final one giving all the warning flags. But it worked. Thank you @ogghead . I can do some experimentation and see how things go with this setup.

## Not closing issue

I am **not** marking this issue as closed because the method to get things working did not match the Readme. For whatever reason, when I followed the readme steps, I was building on v0.2.93. Perhaps the template requires further updating? Perhaps there is something with Spin? Perhaps there is dark magic at work. Regardless, something needs updating, even if that thing is only the readme adding a line to do the cargo command provided by @ogghead .

**Comment by ogghead** [comment]
Glad to hear that worked! On the observation of multiple compilation cycles, I suspect this is caused because the command Spin runs to build your Leptos component is `cargo leptos build --release && LEPTOS_OUTPUT_NAME=test cargo build --lib --target wasm32-wasi --release --no-default-features --features ssr` which performs two builds.

Agreed that updating the README to list the command to install cargo-leptos from GitHub would be ideal -- is that something you might be interested in creating a PR for?

**Comment by ogghead** [comment]
With Leptos 0.7 release, the README now specifies an exact version of `cargo-leptos` to install and includes a section on `wasm-bindgen` mismatches. @dimitrilw I think this issue can be resolved
